localstack/aws/handlers/internal_requests.py

```diff
@@ -0,0 +1,28 @@
+import logging
+
+from localstack.http import Response
+
+from ..api import RequestContext
+from ..chain import Handler, HandlerChain
+from ..connect import INTERNAL_REQUEST_PARAMS_HEADER, load_dto
+
+LOG = logging.getLogger(__name__)
+
+
+class InternalRequestParamsEnricher(Handler):
+    """
+    This handler sets the internal call DTO in the request context.
+
+    Important: This must be invoked after account and region enrichers because
+    it may override them.
+    """
+
+    def __call__(self, chain: HandlerChain, context: RequestContext, response: Response):
+        if header := context.request.headers.get(INTERNAL_REQUEST_PARAMS_HEADER):
+            try:
+                dto = load_dto(header)
```

Comment:
we could consider making this immutable

```suggestion
                dto = MappingProxyType(load_dto(header))
```